R/learner_xgboost_surv_xgboost_cox.R

```diff
@@ -0,0 +1,200 @@
+#' @title Extreme Gradient Boosting Cox Survival Learner
+#' @author bblodfon
+#' @name mlr_learners_surv.xgboost.cox
+#'
+#' @description
+#' eXtreme Gradient Boosting regression using a **Cox Proportional Hazards**
+#' objective.
+#' Calls [xgboost::xgb.train()] from package \CRANpkg{xgboost} with `objective`
```

Comment:
as above, I don't udnerstand why we add different learners for the objectives but then still expose the parameter

Reply:
also, can you remind me why we are refactoring this into two learners (I remember there is a good reason but I don't remember what it was ...)

Reply:
I couldn't make it work with just one learner, see a comment I wrote above, where I setup a `surv.xgboost.cox` learner with an `objective` parameter that pretty much changes the learner type which is confusing (so a Cox learner in name, but underneath an AFT fitting and so on)

Reply:
ok, but see my comment on mattermost